plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureClientProvider.java

```diff
@@ -150,7 +150,6 @@ private static ByteBufAllocator createByteBufAllocator() {
         int tinyCacheSize = PooledByteBufAllocator.defaultTinyCacheSize();
         int smallCacheSize = PooledByteBufAllocator.defaultSmallCacheSize();
         int normalCacheSize = PooledByteBufAllocator.defaultNormalCacheSize();
-        boolean useCacheForAllThreads = PooledByteBufAllocator.defaultUseCacheForAllThreads();
 
         return new PooledByteBufAllocator(false,
             nHeapArena,
```

Comment:
I think that we can reduce those too, wdyt @original-brownbear?

Reply:
Yea, can't we just do `1` maybe when we only use one thread now anyway?